internal/core/domain/vault_model.go

```diff
@@ -37,30 +38,98 @@ type AddressInfo struct {
 	DerivationPath string
 }
 
+// IMnemonicStore defines the required methods to override the default
+// storage of the plaintext mnemonic once Unlocking a Vault.
+// At the moment this is achieved by storing the mnemonic in the in-memory
+// pkg/config store. Soon this will be changed since this package shouldn't
+// depend on config.
+type IMnemonicStore interface {
+	Set(mnemonic string)
+	Unset()
+	IsSet() bool
+	Get() []string
+}
+
+// TOOD: move this outside the domain to decouple from config and use an
+// equivalent alternative.
+type configStore struct{}
+
+func (c configStore) Set(mnemonic string) {
+	config.Set(config.MnemonicKey, mnemonic)
+}
+
+func (c configStore) Unset() {
+	config.Set(config.MnemonicKey, nil)
+}
+
+func (c configStore) IsSet() bool {
+	return len(config.GetMnemonic()) > 0
+}
+
+func (c configStore) Get() []string {
+	return config.GetMnemonic()
+}
+
+// IEncrypter defines the required methods to override the default encryption
+// performed through pkg/wallet package.
+type IEncrypter interface {
```

Comment:
I tried to 'google' bit about adding Prefix 'I' for interfaces, there are different views, I personally dont like it and in this specific place I would use only Encrypter, but there are cases when there is sturct and intercafe with same name and then using 'I' prefix helps, but even in that way I would try  to find some better name.
However dont have clear opinion on this.

Reply:
Not saying one way is better that other, but style changes should be done in a dedicated PR, not mixed with functional/testing stuff.